File: android/mobilepasssdk/src/main/java/com/armongate/mobilepasssdk/model/response/ResponseHandshake.java
package com.armongate.mobilepasssdk.model.response;

/**
 * Response model for handshake endpoint
 * Contains server-generated nonce for request signing
 */
public class ResponseHandshake {
    /**
     * Indicates if handshake was successful
     */
    public boolean ok;

    /**
     * Server-generated nonce used in all request signatures during this session
     */
    public String serverNonce;

    /**
     * Time-to-live for ephemeral key in seconds (e.g., 120)
     * Key should be renewed after this period
     */
    public Long ephemeralTTL;

    /**
     * Error message if handshake failed
     */
    public String error;
}

File: ios/MobilePassSDK/Model/Response/ResponseHandshake.swift
//
//  ResponseHandshake.swift
//  MobilePassSDK
//
//  Response model for handshake completion
//

import Foundation

struct ResponseHandshake: Codable {
    /// Success indicator
    let ok: Bool
    
    /// Server-generated nonce (tied to ephemeral key session)
    let serverNonce: String?
    
    /// Ephemeral key TTL in seconds (typically 120)
    let ephemeralTTL: Int?
    
    /// Optional error message
    let error: String?
}
